Changelog for 10.24.0 (#16888)
MetaMask/metamask-extension: CHANGELOG.md

- Replace the address in SignTypedData_v4 signatures with a 'Verify contract details' link ([#16191](https://github.com/MetaMask/metamask-extension/pull/16191))
- Bump `@metamask/design-tokens` from `1.11.0` to `1.11.1` ([#16764](https://github.com/MetaMask/metamask-extension/pull/16764))
- Bump `@metamask/design-tokens` from `1.9.0` to `1.11.0` ([#16515](https://github.com/MetaMask/metamask-extension/pull/16515))
- Update `@metamask/controllers` to `v33` ([#16493](https://github.com/MetaMask/metamask-extension/pull/16493))
- Migrate from deprecated `@metamask/controllers`to the new controller packages ([#16547](https://github.com/MetaMask/metamask-extension/pull/16547))
- Swaps: ensure 0% slippage on Arbitrum for wrapping/unwrapping ETH (ETH -> WETH or WETH -> ETH) ([#16778](https://github.com/MetaMask/metamask-extension/pull/16778))
- [Beta] Update to sentence case ([#16590](https://github.com/MetaMask/metamask-extension/pull/16590))
- [Beta] Updating welcome message ([#16489](https://github.com/MetaMask/metamask-extension/pull/16489))

> - [FLASK] `snaps-monorepo@0.25.0` ([#16673](https://github.com/MetaMask/metamask-extension/pull/16673))

Reviewer: ```suggestion
- [FLASK] **BREAKING:** Snaps are now required to request the `endowment:rpc` permission to receive RPC requests on `onRpcRequest` ([#16673](https://github.com/MetaMask/metamask-extension/pull/16673))
  - Snaps must specify if they want to receive RPC requests from dapps or snaps using the following permission `endowment:rpc: { dapps: true, snaps: true }`
```

> - [FLASK] `snaps-monorepo@0.24.1` ([#16525](https://github.com/MetaMask/metamask-extension/pull/16525))

Reviewer: ```suggestion
- [FLASK] **BREAKING:** Removed `wallet_enable` and `wallet_installSnaps` in favor of `wallet_requestSnaps` ([#16525](https://github.com/MetaMask/metamask-extension/pull/16525))
- [FLASK] **BREAKING:** Removed `snap_getAppKey` ([#16525](https://github.com/MetaMask/metamask-extension/pull/16525))
  - Snaps that need snap-specific entropy can use `snap_getEntropy` instead
- [FLASK] **BREAKING:** The `wallet` global exposed to Snaps has been replaced with two new globals: `snap` and `ethereum` ([#16525](https://github.com/MetaMask/metamask-extension/pull/16525))
  - `ethereum` is an EIP-1193 provider and can be accessed by requesting the `endowment:ethereum-provider` permission
  - `snap` is always exposed and can be used to access snap specific functions using `snap.request()`
- [FLASK] **BREAKING:** Named parameters are now required in all Snaps RPC methods ([#16525](https://github.com/MetaMask/metamask-extension/pull/16525))
```

Reviewer: Let me know if you need help categorizing these 😅

- Fix unable to proceed with tx when there is "Insufficient funds for gas", by adding the estimated gas fee section on the Send screen ([#14634](https://github.com/MetaMask/metamask-extension/pull/14634))
- Fix Add Contact by disabling the Save button if the input fields are empty ([#16233](https://github.com/MetaMask/metamask-extension/pull/16233))
- Fix Token Detection displaying a token without balance, by updating the user state `tokensChainsCache` from array to object ([#16535](https://github.com/MetaMask/metamask-extension/pull/16535))
- Fix message not rendering properly on Sign Typed Data v4 screen, by supporting nested arrays and arrays with defined length ([#16552](https://github.com/MetaMask/metamask-extension/pull/16552))
- Fix German translation for `lightTheme` ([#16517](https://github.com/MetaMask/metamask-extension/pull/16517))
- [FLASK] Clear notification state on restore ([#16503](https://github.com/MetaMask/metamask-extension/pull/16503))
- [FLASK] Fix a crash that happens after snap install ([#16526](https://github.com/MetaMask/metamask-extension/pull/16526))

> - [FLASK] Bump snaps iframe-execution-environment ([#16932](https://github.com/MetaMask/metamask-extension/pull/16932))

Reviewer: ```suggestion
- [FLASK] Fix usage of wrong `ethereum` global for `ethereum` endowment  ([#16932](https://github.com/MetaMask/metamask-extension/pull/16932))
```
